src/reconcile/rec_col.c

```diff
@@ -882,6 +873,15 @@ __wt_rec_col_var(
                   ((deleted && last.deleted) ||
                       (!deleted && !last.deleted && last.value->size == size &&
                         memcmp(last.value->data, data, size) == 0))) {
+                    /*
+                     * The start time pair for deleted keys must be (WT_TS_NONE, WT_TXN_NONE) and
+                     * stop time pair must be (WT_TS_MAX, WT_TXN_MAX) since we no longer select
+                     * tombstone to write to disk and the deletion of the keys must be globally
+                     * visible.
+                     */
+                    WT_ASSERT(session, (!deleted && !last.deleted) ||
+                        (last.start_ts == WT_TS_NONE && last.start_txn == WT_TXN_NONE &&
+                                         last.stop_ts == WT_TS_MAX && last.stop_txn == WT_TXN_MAX));
```

Comment:
Is this true, even when we took a deleted record from an update? So, the user does `WT_CURSOR.remove`, that puts an update on the chain that is a "deleted" record. Will `last.XXX` be `NONE/MAX`?

Reply:
That currently is reconciled as a normal record on the page with a stop timestamp. We haven't decided how to remove them from the disk image yet.

Reply:
If we can remove the record from the disk image, then the deletion must be globally visible. We can set its start time pair to (0, 0) and stop time pair to (MAX, MAX).